tools/clang/test/CodeGenSPIRV/method.buffer.load.offset.error.hlsl

```diff
@@ -0,0 +1,7 @@
+// RUN: not %dxc -T ps_6_2 -E main -fcgl -spirv %s 2>&1 | FileCheck %s
+
+Texture2D SourceTexture;
+void main(in float4 SvPosition : SV_Position, out float4 OutColor : SV_Target0) {
+    // CHECK: error: Offsets to texture access operations must be immediate values.
+    OutColor = SourceTexture.Load(SvPosition.x, SvPosition.y);
```

Comment:
DXIL behavior seems slight different than what the error says:
- this code does compile with `-fcgl`
- this code throws an error with any other optimization level.

The following code works:
```
float foo() {
  return 0;
}

Texture2D SourceTexture;
void main(in float4 SvPosition : SV_Position, out float4 OutColor : SV_Target0) {
  // CHECK: error: Offsets to texture access operations must be immediate values.
  OutColor = SourceTexture.Load(SvPosition.x, foo());
}
```

Seems like they don't require an immediate, but something that evaluates to a constant, and always legalize the code if a `Load` is used. (The snippets also compiles with `-O0`).

Reply:
> DXIL behavior seems slight different than what the error says:
> 
> * this code does compile with `-fcgl`
> * this code throws an error with any other optimization level.

It looks like the pass in which DXIL does this validation is skipped with `-fcgl`, so I wouldn't really consider it relevant.

> The following code works:
> 
> ```
> float foo() {
>   return 0;
> }
> 
> Texture2D SourceTexture;
> void main(in float4 SvPosition : SV_Position, out float4 OutColor : SV_Target0) {
>   // CHECK: error: Offsets to texture access operations must be immediate values.
>   OutColor = SourceTexture.Load(SvPosition.x, foo());
> }
> ```
>
> Seems like they don't require an immediate, but something that evaluates to a constant, and always legalize the code if a `Load` is used. (The snippets also compiles with `-O0`).

That's correct. In reality both backends attempt to evaluate the expression to a constant int, and that behavior likely has some edge cases that are slightly different. Since we don't have any formal `constexpr` though, it's a bit vague to define what's accepted so I think it's alright to keep the error message consistent and more narrow than what we actually support.